client/src/js/components/bhDepotDropdown.js

```diff
@@ -35,6 +36,13 @@ function bhDepotController(Depot, AppCache, Notify) {
   $ctrl.select = function select(option) {
     $ctrl.selection = option;
     cache.selection = $ctrl.selection;
-    $ctrl.onSelect({ depot: $ctrl.selection });
+    $ctrl.onSelect({ depot : $ctrl.selection });
   };
+
+  function cachedDepotExist() {
+    if (!cache.selection) { return false; }
+    return $ctrl.depots.map(function (item) {
```

Comment:
Hmm... while this looks cool, it is _really_ hard to read.  You probably want a `filter`.  It may even be faster.
```js
// returns true if the cached uuid matches a depot uuid
return $ctrl.depots.filter(function (depot) {
  return depot.uuid === cache.selection.uuid;
});
```